--- a/src/SDUtils.hpp
+++ b/src/SDUtils.hpp
@@ -392,7 +392,8 @@
 
 // Encodes an RGB byte array into PNG format
 inline std::vector<uint8_t> encodePNG(const std::vector<uint8_t> &rgb_data,
-                                      int width, int height) {
+                                      int width, int height,
+                                      int channels = 3) {
   std::vector<uint8_t> png_buffer;
 
   stbi_write_png_to_func(
@@ -401,7 +402,7 @@
         buffer.insert(buffer.end(), static_cast<uint8_t *>(data),
                       static_cast<uint8_t *>(data) + size);
       },
-      &png_buffer, width, height, 3, rgb_data.data(), width * 3);
+      &png_buffer, width, height, channels, rgb_data.data(), width * channels);
 
   return png_buffer;
 }
@@ -409,7 +410,7 @@
 // Encodes an RGB byte array into JPEG format (quality adjustable)
 inline std::vector<uint8_t> encodeJPEG(const std::vector<uint8_t> &rgb_data,
                                        int width, int height,
-                                       int quality = 95) {
+                                       int quality = 95, int channels = 3) {
   std::vector<uint8_t> jpeg_buffer;
 
   stbi_write_jpg_to_func(
@@ -418,7 +419,7 @@
         buffer.insert(buffer.end(), static_cast<uint8_t *>(data),
                       static_cast<uint8_t *>(data) + size);
       },
-      &jpeg_buffer, width, height, 3, rgb_data.data(), quality);
+      &jpeg_buffer, width, height, channels, rgb_data.data(), quality);
 
   return jpeg_buffer;
 }
